pylon-core/src/main/kotlin/io/github/pylonmc/pylon/core/entity/display/transform/Translation.kt

```diff
@@ -0,0 +1,18 @@
+package io.github.pylonmc.pylon.core.entity.display.transform
+
+import org.joml.Matrix4f
+import org.joml.Vector3d
+import org.joml.Vector3f
+
+class Translation(private val translation: Vector3f) : TransformComponent {
+
+    constructor(translation: Vector3d): this(Vector3f(translation))
+
+    constructor(x: Float, y: Float, z: Float): this(Vector3f(x, y, z))
+
+    constructor(x: Double, y: Double, z: Double): this(Vector3d(x, y, z))
+
+    override fun apply(matrix: Matrix4f) {
+        matrix.mul(Matrix4f().translate(translation))
```

Comment:
```suggestion
        matrix.translate(translation)
```
I honestly have no clue what you were cooking here

Reply:
There is a reason that this is written the way it is. I have no idea what that reason is, but this is what was in display model lib and frankly I do not fancy spending another 3 hours trying to figure out what the right magic incantation is to get my block displays to have the right transformations

Reply:
example translating (10, 10, 5) by (-2, -5, 1):
Matrix4f() creates (1,1,1,1)
then translate by (-2, -5, 1) gives (-1, -4, 2, 1)
then multiply (10, 10, 5) by (-1, -4, 2, 1) gives (-10, -40, 10),
but yknow if its copy pasted I'm not gonna question it ig but that is meth not math